src/lib/pubkey/ec_group/ec_group.cpp

```diff
@@ -370,11 +370,25 @@ EC_Group::EC_Group(const OID& oid,
                    const BigInt& base_y,
                    const BigInt& order) {
    BOTAN_ARG_CHECK(oid.has_value(), "An OID is required for creating an EC_Group");
+   // TODO(Botan4) change lower bound to 192 bits
    BOTAN_ARG_CHECK(p.bits() >= 128, "EC_Group p too small");
    BOTAN_ARG_CHECK(p.bits() <= 521, "EC_Group p too large");
 
    if(p.bits() == 521) {
-      BOTAN_ARG_CHECK(p == BigInt::power_of_2(521) - 1, "EC_Group with p of 521 bits must be 2**521-1");
+      const auto p521 = BigInt::power_of_2(521) - 1;
+      BOTAN_ARG_CHECK(p == p521, "EC_Group with p of 521 bits must be 2**521-1");
+   } else if(p.bits() == 239) {
+      const auto x962_p239 = []() {
+         BigInt p239;
+         for(size_t i = 0; i != 239; ++i) {
+            if(i < 47 || ((i >= 94) && (i != 143))) {
+               p239.set_bit(i);
+            }
+         }
+         return p239;
```

Comment:
This strikes me like a good place to use https://github.com/randombit/botan/pull/4475, no?

Reply:
Maybe? This code only gets executed if you are actually trying to register an X9.62 239-bit prime, which I expect to happen ~~~ never. Like there is maybe one user of those curves, and I don't want to leave them completely high and dry, but don't care that much either.

[OTOH this perhaps argues to not including the 239-bit exception at all, and then if it really is an issue in Botan4 they can open an issue and then we can add the exception at the time.]

Reply:
Check.